File: app/src/main/java/me/hachy/routineweek/util/Prefs.java
package me.hachy.routineweek.util;


import android.content.Context;
import android.content.SharedPreferences;

public class Prefs {

    private static final String PREF_NAME = "routine_week";
    private static final String COLOR = "color";
    private Context context;

    public Prefs(Context context) {
        this.context = context;
    }

    public void setTagColorIdx(int color) {
        SharedPreferences preferences = context.getSharedPreferences(PREF_NAME, Context.MODE_PRIVATE);
        SharedPreferences.Editor editor = preferences.edit();
        editor.putInt(COLOR, color);
        editor.apply();
    }

    public int getTagColorIdx() {
        SharedPreferences preferences = context.getSharedPreferences(PREF_NAME, Context.MODE_PRIVATE);
        return preferences.getInt(COLOR, 0);
    }

}

File: app/src/main/java/me/hachy/routineweek/util/Prefs.kt
package me.hachy.routineweek.util


import android.content.Context

class Prefs(private val context: Context) {

    var tagColorIdx: Int
        get() {
            val preferences = context.getSharedPreferences(PREF_NAME, Context.MODE_PRIVATE)
            return preferences.getInt(COLOR, 0)
        }
        set(color) {
            val preferences = context.getSharedPreferences(PREF_NAME, Context.MODE_PRIVATE)
            val editor = preferences.edit()
            editor.putInt(COLOR, color)
            editor.apply()
        }

    companion object {

        private val PREF_NAME = "routine_week"
        private val COLOR = "color"
    }

}
